frontend/src/components/MainComponent.js

```diff
@@ -0,0 +1,34 @@
+import React, { useContext, useEffect } from 'react';
+import TableComponent from './TableComponent';
+import AddModal from './AddModal';
+import HeaderComponent from './HeaderComponent';
+import { userContext } from '../context/userContext';
+import { baseUrl } from '../shared/baseUrl';
+
+const Main = () => {
+	const { users, getUsers, addUser, deleteUser } = useContext(userContext);
+
+	useEffect(() => {
+		getUsers();
+	}, [])
```

Comment:
Didn't you face a problem of continuously calling getUsers()

Reply:
No ayye because I used the [ ] as a second parameter it is acting as a componentDidMount lifecycle method. I checked with the json-server and it doesnt continiously GET the data